app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,22 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
```

Comment:
giải thích cho a cách toán tử `||=` hoạt động.
và tại sao ở đây lại dùng biến instance?

Reply:
cái ||= nó sẽ hoạt động theo kiểu nếu cái @current_user này nó nil  thì sẽ gán giá trị vào cái @current_user. sự dụng biết instance ở đây vì nếu giá trị không xác định thì nó sẽ trả về nil.

Reply:
tìm hiểu lại sao lại dùng biến instance ở đây nhé.

Reply:
biến instance để cách phương thức khác có thể sử dụng được 
 
 def logged_in?
    current_user.present?
  end

  def log_out
    session.delete :user_id
    @current_user = nil
  end

Reply:
nếu không phải biến instance thì các phương thức khác vẫn sử dụng bình thường được hàm current_user đấy thôi.
thử nhớ lại phạm vi sử dụng của biến @instance, về cách hoạt động của toán tử `||=`, về tần suất sử dụng hàm `current_user` và xâu chuỗi với nhau xem nào

Reply:
sử dụng biến instance vì nó có thể tái sử dụng, còn biến local thì không, nếu dùng biến local thì nó sẽ làm chậm hiệu suất của ứng dụng. Trong cái file này thì nếu current_user dùng biến local thì mỗi lần dùng đến đều phải gọi cái này User.find_by id: session[:user_id]